D1673/P1673.md

```diff
@@ -6764,82 +6534,59 @@ public:
     offset(data_handle_type p, size_t i) const
       noexcept(noexcept(@_nested-accessor_@.offset(p, i)));
 
-  constexpr Accessor nested_accessor() const;
+  constexpr const Accessor& nested_accessor() const;
 };
 ```
 
-[2]{.pnum} `Accessor` meets the `mdspan` accessor policy requirements
-**[mdspan.accessor.reqmts]**.
+[2]{.pnum} *Mandates:*
 
-```c++
-using reference = /* see below */;
-```
+  * [2.1]{.pnum} `is_copy_constructible_v<element_type>` is `true`.
 
-[3]{.pnum} This names
+  * [2.2]{.pnum} `Accessor` meets the accessor policy requirements **[mdspan.accessor.reqmts]**.
 
-* [3.1]{.pnum} `typename Accessor::reference`,
-    if `remove_cv_t<typename Accessor::element_type>` is an arithmetic type;
+  * [2.3]{.pnum} _`scalable`_`<ScalingFactor, typename Accessor::element_type>` is `true`. 
 
-* [3.2]{.pnum} otherwise,
-    _`conjugated-scalar`_`<typename Accessor::reference, remove_cv_t<typename Accessor::element_type>>`.
 
 ```c++
-using element_type = /* see below */
+constexpr conjugate_accessor(const Accessor& acc);
 ```
 
-[4]{.pnum} Let the exposition-only type `pre-const-element-type` name
-
-* [4.1]{.pnum} `typename Accessor::element_type`,
-    if `remove_cv<typename Accessor::element_type>` is an arithmetic type;
-
-* [4.2]{.pnum} otherwise, `reference::value_type`.
-
-[5]{.pnum} Then, `element_type` names `add_const_t<`_`pre-const-element-type`_`>`.
-
-<i>[Note:</i> `conjugate_accessor` provides read-only access. <i>-- end note]</i>
+[3]{.pnum} *Effects:* Direct non-list-initializes
+_`nested-accessor`_ with `acc`.
 
 ```c++
-using offset_policy = /* see below */;
+template<class OtherScalingFactor, class OtherNestedAccessor>
+  constexpr conjugate_accessor(const conjugate_accessor<OtherScalingFactor, OtherNestedAccessor>& other);
 ```
 
-[6]{.pnum} This names
-
-* [6.1]{.pnum} `typename Accessor::offset_policy`,
-    if `remove_cv_t<typename Accessor::element_type>` is an arithmetic type;
+[4]{.pnum} *Constraints:* `is_constructible_v<`_`nested-accessor`_`, const OtherNestedAccessor&>` is `true`, and
 
-* [6.2]{.pnum} otherwise, `conjugate_accessor<typename Accessor::offset_policy>`.
-
-```c++
-constexpr conjugate_accessor(Accessor a);
-```
-
-[7]{.pnum} *Effects:* Direct non-list-initializes
-_`nested-accessor`_ with `a`.
+[5]{.pnum} *Effects:* direct non-list-initializes _`nested-accessor`_ with `other.nested_accessor()`.
 
 ```c++
 constexpr reference access(data_handle_type p, size_t i) const
   noexcept(noexcept(reference(@_nested-accessor_@.access(p, i))));
 ```
 
-[8]{.pnum} *Returns:*
-`reference(`_`nested-accessor`_`.access(p, i))`
+[6]{.pnum} *Returns:*
+_`conj-if-needed`_`(`_`nested-accessor`_`.access(p, i))`
```

Comment:
I think we may need to cast the result of the inner `access` call to `Accessor::element_type`.  The issue is that we define `element_type` as

```
using element_type = decltype(@_conj-if-needed_@(declval<Accessor::element_type>()));
```

Users could, in theory, provide an ADL-accessible `conj(Accessor::reference)` with a different return type than the ADL-accessible `conj(Accessor::element_type)`.  Casting the result to `Accessor::element_type` would make it unambiguous which `conj` is called.

Alternately, if `conj(Accessor::reference)` does lazy computation (e.g., deferring the expensive load of a proxy reference) while `conj(Accessor::element_type)` does eager computation, we might actually want `conj(Accessor::reference)` here.

In either case, we should make the definition of `element_type` consistent with the specification of the `access` function.

Reply:
Yeah of course, I had that in mind but forgot.